prov/efa/src/rdm/rxr_env.c

```diff
@@ -281,39 +248,42 @@ void rxr_env_initialize()
 /**
  * @brief Fetch the value of the environment variable FI_EFA_USE_DEVICE_RDMA
  *
- * This function fetches the change in the value of FI_EFA_USE_DEVICE_RDMA
- * if any, after it was defined and set in rxr_env_define_use_device_rdma() 
- * 
+ * This function fetches the value of FI_EFA_USE_DEVICE_RDMA from the
+ * user's environment.
+ *
+ * The return value is not boolean.  It may be 1, 0, or -FI_ENODATA.
+ *
+ * Environments with non-boolean (1, true, yes, on) values for the variable
+ * will print to stderr and abort the program.
+ *
  * @return		0 - If FI_EFA_USE_DEVICE_RDMA is set to 0/false/no/off
- * 			1 - If FI_EFA_USE_DEVICE_RDMA is set to 1/true/yes/on
- * 			If FI_EFA_USE_DEVICE_RDMA is not set, then the value
- * 			returned from rxr_env_get_default_use_device_rdma()
- * 			will be the return value of this function.
+ *			1 - If FI_EFA_USE_DEVICE_RDMA is set to 1/true/yes/on
+ *			-FI_ENODATA - if FI_EFA_USE_DEVICE_RDMA is not set.
```

Comment:
I think use return value both for result and return code is problematic.

Do we really need a function to both return the value of the envar and the value of the envar?

Can we just have a function `rxr_env_has_use_device_rdma()`? and use the other two functions to

Reply:
I think our logic could be:

`if env_has_device_rdma() && env_get_rdma()`

or it could be

`if env_get_rdma() == 1`

I like the second one.

Reply:
Using two boolean functions now.